### ISSUE TITLE
GetAuthorizationRequestUrlParameterBuilder.ExecuteAsync() method returns wrong datatype

### ISSUE THREAD

**Issue by Rick-Kirkham** [AzureAD/microsoft-authentication-library-for-dotnet#1184]
**Which Version of MSAL are you using ?**
Latest 3.x.x

**Platform**
net45

**What authentication flow has the issue?**
* Desktop / Mobile
    * [ ] Interactive
    * [ ] Integrated Windows Auth
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web App
    * [x ] Authorization code
    * [ ] OBO
* Web API
    * [ ] OBO

**Is this a new or existing app?**
c. This is a new app 

**Repro**

```csharp
string[] graphScopes = { "Files.Read.All", "User.Read" };

ConfidentialClientApplicationBuilder clientBuilder = ConfidentialClientApplicationBuilder.Create(Settings.AzureADClientId);
ConfidentialClientApplication clientApp = (ConfidentialClientApplication)clientBuilder.Build();

var urlBuilder = clientApp.GetAuthorizationRequestUrl(graphScopes);
urlBuilder.WithRedirectUri(loginRedirectUri.ToString());
urlBuilder.WithAuthority(Settings.AzureADAuthority);
var authUrl = await urlBuilder.ExecuteAsync();
authUrl. ????
```

**Expected behavior**
According to  [ConfidentialClientApplication.GetAuthorizationRequestUrl](https://docs.microsoft.com/en-us/dotnet/api/microsoft.identity.client.confidentialclientapplication.getauthorizationrequesturl?view=azure-dotnet), the method:

> Computes the URL of the authorization request letting the user sign-in and consent to the application accessing specific scopes in the user's name. The URL targets the /authorize endpoint of the authority configured in the application. 

So, in the code in **Repro** above, the object returned by `GetAuthorizationRequestUrlParameterBuilder.ExecuteAsync()` should return a URL or an object with a property that is a URL. 

**Actual behavior**
The `GetAuthorizationRequestUrlParameterBuilder.ExecuteAsync()` method returns an `AuthorizationResult` which is not a URL and has no property with a URL in it.

**Comment by jmprieur** [comment]
@bgavrilMS fyi

**Comment by bgavrilMS** [comment]
Fix will be available in 4.0 release

**Comment by jennyf19** [comment]
@Rick-Kirkham Fixed in [MSAL 4.0.0 release](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.0.0)

**Comment by Rick-Kirkham** [comment]
@jennyf19 Has 4.0 officially GAed or is it still preview/beta?

**Comment by jennyf19** [comment]
@Rick-Kirkham We actually GA'ed w/3.0.8, so 4.0 is also GA.

**Comment by Rick-Kirkham** [comment]
??? You GAed two versions at once? Why? I can't make sense of that.

**Comment by jennyf19** [comment]
@Rick-Kirkham we GA'ed w/3.0.8 and 4.0 is not in preview or beta. We [GA'ed at the beginning of May](https://developer.microsoft.com/en-us/identity/blogs/msal-net-is-now-generally-available/).

**Comment by Rick-Kirkham** [comment]
@jennyf19 Just last week I installed the NuGet package for MSAL in a Visual Studio project. The most recent version available was 3.0.8. There was no 4.0.

**Comment by Rick-Kirkham** [comment]
@jmprieur  
I see. Thanks. When will 4.0 be available in NuGet?

**Comment by jennyf19** [comment]
@Rick-Kirkham It's available now [msal 4.0.0](https://www.nuget.org/packages/Microsoft.Identity.Client/4.0.0)

**Comment by Rick-Kirkham** [comment]
Got it. Thanks.
